src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -88,23 +91,35 @@ protected OtpUser getUserProfile(Auth0UserProfile profile) {
      * HTTP endpoint to send an SMS text to an {@link OtpUser}'s phone number with a verification code. This is used
      * during user signup (or if a user wishes to change their notification preferences to use a new un-verified phone
      * number).
+     * Before sending a SMS request, this endpoint also saves the submitted phone number, marks it as not verified.
      */
     public VerificationResult sendVerificationText(Request req, Response res) {
         OtpUser otpUser = getEntityForId(req, res);
-        if (otpUser.phoneNumber == null) {
-            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, "User must have valid phone number to verify.");
+        // Get phone number from the path param.
+        String phoneNumber = req.params(PHONE_PARAM);
```

Comment:
I'm wondering now for before we send the verification text, if there is a Twilio service we can call to check whether the phone number is already verified by Twilio?

Reply:
Twilio offers [caller ID management for outgoing calls](https://support.twilio.com/hc/en-us/articles/223180048-Adding-a-Verified-Phone-Number-or-Caller-ID-with-Twilio), but that is not quite what you are talking about.

Reply:
Actually it is, "outgoing" meaning the destination number a message goes out to... Do you want to address it in this PR or another one?

Reply:
I was just curious what was out there, no need to address.